src/NelsonSiegelSvensson.jl

```diff
@@ -0,0 +1,179 @@
+abstract type ParametricModel end
+
+"""
+    NelsonSiegel(β₀, β₁, β₂, τ₁)
+
+Parameters of Nelson and Siegel (1987) parametric model. 
+
+# Examples
+
+```julia-repl
+julia> β₀, β₁, β₂, τ₁ = 0.6, -1.2, -1.9, 3.0
+julia> nsm = NelsonSiegelSvensson.NelsonSiegel.(β₀, β₁, β₂, τ₁)
+```
+"""
+struct NelsonSiegel <: ParametricModel
+    β₀
+    β₁
+    β₂
+    τ₁
+    #r² no need
+
+    #function NelsonSiegel(β₀, β₁, β₂, τ₁, r²)
+    function NelsonSiegel(β₀, β₁, β₂, τ₁)
+        #(τ₁ <= 0 || r² < 0 || r² > 1) && throw(DomainError("Wrong parameter ranges"))
+        (τ₁ <= 0) && throw(DomainError("Wrong parameter ranges"))
+        #return new(β₀, β₁, β₂, τ₁, r²)
+        return new(β₀, β₁, β₂, τ₁)
+    end
+end
+
+"""
+    NelsonSiegelSvensson(β₀, β₁, β₂, β₃, τ₁, τ₂)
+
+Parameters of Svensson (1994) parametric model. 
+
+# Examples
+
+```julia-repl
+julia> β₀, β₁, β₂, β₃, τ₁, τ₂ = 0.6, -1.2, -2.1, 3.0, 1.5
+julia> nssm = NelsonSiegelSvensson.NelsonSiegelSvensson.(β₀, β₁, β₂, β₃, τ₁, τ₂)
+```
+"""
+struct NelsonSiegelSvensson <: ParametricModel
+    β₀
+    β₁
+    β₂
+    β₃
+    τ₁
+    τ₂
+    #r² no need
+
+    #function NelsonSiegelSvensson(β₀, β₁, β₂, β₃, τ₁, τ₂, r²)
+    function NelsonSiegelSvensson(β₀, β₁, β₂, β₃, τ₁, τ₂)
+        #(τ₁ <= 0 || τ₂ <= 0 || r² < 0 || r² > 1) && throw(DomainError("Wrong parameter ranges"))
+        (τ₁ <= 0 || τ₂ <= 0) && throw(DomainError("Wrong parameter ranges"))
+        #return new(β₀, β₁, β₂, β₃, τ₁, τ₂, r²)
+        return new(β₀, β₁, β₂, β₃, τ₁, τ₂)
+    end
+end
+
+Base.zero(ns::NelsonSiegel, t) = ns.β₀ .+ ns.β₁ .* (1.0 .- exp.(-t ./ ns.τ₁)) ./ (t ./ ns.τ₁) .+ ns.β₂ .* ((1.0 .- exp.(-t ./ ns.τ₁)) ./ (t ./ ns.τ₁) .- exp.(-t ./ ns.τ₁))
```

Comment:
`Base.zero` should return a `Rate`, and for Nelson-Siegel and friends, that should be a continuous compounding rate. More concretely, the right-hand-side should be wrapped in a `Continuous(   )`.

Reply:
But then in the definition of `discount`, `zero` has to be converted back into a real number (rather than a `Rate`).